shared/actions/fs.js

```diff
@@ -96,13 +96,18 @@ function* download(action: FsGen.DownloadPayload): Saga.SagaGenerator<any, any>
     },
   })
 
-  // Fake out progress until we have the real thing.
-  // TODO: have the real thing.
-  const total = 6
-  for (let progress = 0; progress < total - 1; ++progress) {
+  let completePortion = 0
+  while (completePortion < 1) {
     yield Saga.delay(500)
-    yield Saga.call(RPCTypes.SimpleFSSimpleFSCheckRpcPromise, {opID})
-    yield Saga.put(FsGen.createFileTransferProgress({key, completePortion: progress / total}))
+    const progress = yield Saga.call(RPCTypes.SimpleFSSimpleFSCheckRpcPromise, {opID})
+    completePortion = progress.bytesWritten / progress.bytesTotal
+    yield Saga.put(
+      FsGen.createFileTransferProgress({
+        key,
+        endEstimate: progress.endEstimate,
+        completePortion,
+      })
+    )
```

Comment:
I'm a little worried about using `completePortion` for the comparator, since it's likely to experience floating point error.

I'd rather this be:
```javascript
let progress
do {
  progress = yield Saga.call(RPCTypes.SimpleFSSimpleFSCheckRpcPromise, {opID})
  yield Saga.put(
    FsGen.createFileTransferProgress({
      key,
      endEstimate: progress.endEstimate,
      completePortion: progress.bytesWritten / progress.bytesTotal,
    })
  )
} while (progress.bytesWritten < progress.bytesTotal)

Reply:
Hmm good point. Will do thanks!

Reply:
Oh and we might want to put a throttle on the progress checker--we don't need it to spin quickly.

Reply:
This is delayed with 500ms. Is that too short?

Reply:
Oh then that's probably fine. Where is the delay?

Reply:
It's at the beginning of the loop. There's a Saga.delay(500)

Reply:
Oh lol! I missed it because it wasn't green :P